lib/sdk/writer/index.js

```diff
@@ -636,7 +636,13 @@ class ImageWriter extends EventEmitter {
       this.emit('finish', {
         bytesRead: this.bytesRead,
         bytesWritten: this.bytesWritten,
-        checksum: this.checksum
+        checksum: this.checksum,
+        errors: Array.from(this.destinations).map(([ device, dest ]) => {
+          if (dest.error) {
+            dest.error.device = device
+          }
+          return dest.error
+        }).filter(Boolean)
```

Comment:
Why not filter first? 

```js
Array.from(this.destinations).filter((dest) => {
  return dest.error
}).map(([ device, dest ]) => {
  dest.error.device = device
  return dest.error
})
```

Reply:
Also, the `device` property will need adding to `errors.toJSON()` in `lib/shared/errors.js`